src/pubsub/mod_pubsub_db_rdbms.erl

```diff
@@ -766,7 +774,16 @@ item_to_record({NodeIdx, ItemId, CreatedLUser, CreatedLServer, CreatedAt,
 decode_publisher(null) ->
     undefined;
 decode_publisher(Binary) ->
-    jid:from_binary(Binary).
+    jid_from_binary(Binary).
+
+jid_from_binary(Binary) ->
+    Jid = jid:from_binary(Binary),
+    case Jid of
+        #jid{} ->
+            Jid;
+        _ ->
+            error(#{what => jid_from_binary_failed, encoded_jid => Binary})
+    end.
```

Comment:
Probably because jid:from_binary/1 silently returns `error`, and after that that data flows in the system, until it breaks in completely different place, because that place expects a jid.

I would use `jid:from_binary_really/1` here, if there was one though.

Reply:
ok, would remove that code.

Reply:
Right, I see what you mean and why it would be clearer, but I think the proposed solution was too much to be included within this ticket, it's a different task. You could make a backlog item in the "Hall of Shame" epic we have if you want 😛